chat/avatar/src/Avatar/ChatAvatar.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ForwardedRef, forwardRef } from 'react';
+import { useLeafyGreenChatContext } from '@lg-chat/leafygreen-chat-provider';
+
+import { Avatar, Format, getInitials } from '@leafygreen-ui/avatar';
+import { cx } from '@leafygreen-ui/emotion';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { breakpoints } from '@leafygreen-ui/tokens';
+
+import {
+  iconAvatarStyleOverrides,
+  logoAvatarStyleOverrides,
+  textAvatarStyleOverrides,
+} from './ChatAvatar.styles';
+import {
+  type ChatAvatarProps,
+  ChatAvatarSize,
+  ChatAvatarVariant,
+} from './ChatAvatar.types';
+
+export const chatAvatarSizeMap: Record<ChatAvatarSize, number> = {
+  [ChatAvatarSize.Default]: 52,
+  [ChatAvatarSize.Small]: 40,
+};
+
+export const variantToAvatarFormatMap: Record<ChatAvatarVariant, Format> = {
+  [ChatAvatarVariant.Default]: Format.Icon,
+  [ChatAvatarVariant.Mongo]: Format.MongoDB,
+  [ChatAvatarVariant.User]: Format.Text,
+};
+
+export const ChatAvatar = forwardRef(
+  (
+    {
+      variant = ChatAvatarVariant.Default,
+      size: sizeProp,
+      sizeOverride: sizeOverrideProp,
+      darkMode: darkModeProp,
+      name,
+      ...rest
+    }: ChatAvatarProps,
+    fwdRef: ForwardedRef<HTMLDivElement>,
+  ) => {
+    const { theme } = useDarkMode(darkModeProp);
+    const { containerWidth } = useLeafyGreenChatContext();
+    const size =
+      sizeProp || (containerWidth && containerWidth < breakpoints.Tablet)
```

Comment:
If i'm understanding correctly, if the consumer chooses default for the size, it will ignore that and set the size to small?

Reply:
Yeah, I tried to not touch any of the internal logic in the ChatAvatar component